xchange-stream-btcmarkets/src/main/java/info/bitrich/xchangestream/btcmarkets/BTCMarketsStreamingService.java

```diff
@@ -13,57 +13,78 @@
 
 class BTCMarketsStreamingService extends JsonNettyStreamingService {
   static final String CHANNEL_ORDERBOOK = "orderbook";
+  static final String CHANNEL_TICK = "tick";
   static final String CHANNEL_HEARTBEAT = "heartbeat";
   private static final Logger LOG = LoggerFactory.getLogger(BTCMarketsStreamingService.class);
   private final Set<String> subscribedOrderbooks = Sets.newConcurrentHashSet();
+  private final Set<String> subscribedtick = Sets.newConcurrentHashSet();
 
   public BTCMarketsStreamingService(String apiUrl) {
     super(apiUrl);
   }
 
   private BTCMarketsWebSocketSubscribeMessage buildSubscribeMessage() {
     return new BTCMarketsWebSocketSubscribeMessage(
-        new ArrayList<>(subscribedOrderbooks),
-        Lists.newArrayList(CHANNEL_ORDERBOOK, CHANNEL_HEARTBEAT),
-        null,
-        null,
-        null);
+            new ArrayList<>(subscribedtick),
+            Lists.newArrayList(CHANNEL_ORDERBOOK,CHANNEL_HEARTBEAT,CHANNEL_TICK),
+            null,
+            null,
+            null);
   }
 
   @Override
   protected String getChannelNameFromMessage(JsonNode message) {
     final String messageType = message.get("messageType").asText();
-    if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+    if (messageType.startsWith(CHANNEL_TICK)) {
       return messageType + ":" + message.get("marketId").asText();
     }
-    return messageType;
+    else  if (messageType.startsWith(CHANNEL_ORDERBOOK)) {
+      return messageType + ":" + message.get("marketId").asText();
+    }
+    else
+      return messageType;
   }
 
   @Override
   public String getSubscribeMessage(String channelName, Object... args) throws IOException {
-    if (CHANNEL_ORDERBOOK.equals(channelName)) {
-      subscribedOrderbooks.add(args[0].toString());
-      LOG.debug("Now subscribed to orderbooks {}", subscribedOrderbooks);
+    if (CHANNEL_TICK.equals(channelName)) {
```

Comment:
the logic here seems a bit inconsistent, you introduce a new channelName but use the same `Set<String>`  `subscribedtick` for both. I think you need to use `subscribedtick` when you are adding channel ticks and `subscribedOrderbooks` when you are subscribing to orderbooks and when you `buildSubscribeMessage()` you need to pass in the correct set so as to build the subscribe message for the correct one.

Reply:
Fixed!